daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DFDLXFunctions.scala

```diff
@@ -29,6 +30,76 @@ import org.apache.daffodil.processors.unparsers.UState
 import org.apache.daffodil.processors.unparsers.UnparseError
 import org.apache.daffodil.util.Maybe.Nope
 import org.apache.daffodil.util.Maybe.One
+import passera.unsigned.{UInt, ULong}
+
+/**
+ * This is "logical" shift left, meaning it sign-extends a negative signed value.
+ * dfdl:leftShift for signed values does sign-extension, for logical (zero-filling) left shift you must use an unsigned type
```

Comment:
This should be ``dfdlx:left`` shift, missing the x. Also, I guess tehcnically left shifting will keep the sign, but I'm not sure I would call that sign-extending? And logical/artihmetic is he same for signed/unsigned types. So this comment isn't quite accurate.